Index: sources/scripts/user/User.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import { Hand } from \"../cards/Hand.js\";\r\n\r\nclass User {\r\n    constructor(formData) {\r\n        this.username = formData.get('username');\r\n        this.victory = 0;\r\n        this.hand = new Hand();\r\n    }\r\n\r\n    isValid() {\r\n        return this.username;\r\n    }\r\n\r\n    receiveCard(card) {\r\n        this.hand.addCard(card);\r\n    }\r\n\r\n    hadValidHand() {\r\n        return this.hand.isValid();\r\n    }\r\n\r\n    resetHand(){\r\n        this.hand = new Hand();\r\n    }\r\n}\r\n\r\nexport { User };
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- sources/scripts/user/User.js	(revision 4fa8d254a5261a4220029d0348f91941b4a03188)
+++ sources/scripts/user/User.js	(date 1642094567162)
@@ -22,6 +22,7 @@
     resetHand(){
         this.hand = new Hand();
     }
+
 }
 
 export { User };
\ No newline at end of file
Index: sources/scripts/storage/CardStorage.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- sources/scripts/storage/CardStorage.js	(date 1642096805352)
+++ sources/scripts/storage/CardStorage.js	(date 1642096805352)
@@ -0,0 +1,31 @@
+
+class CardStorage {
+
+    static addCardToStorage(cardData) {
+
+        const card = {
+            suit : cardData.suit ,
+            image : cardData.image ,
+            code : cardData.code ,
+            value : cardData.value
+        }
+        const storageLength = window.localStorage.length ;
+        for(const data in card) {
+            const storageIndex = data + "_" + storageLength ;
+            window.localStorage.setItem(storageIndex, card[data]) ;
+        }
+    }
+
+    static clearStorage() { window.localStorage.clear() } ;
+
+    static retrieveCards() {
+        const cards = [] ;
+        const nbCardStored = window.localStorage.length / 4 ;
+        for(let i = 0 ; i <= window.localStorage.length ; i + 4 ) {
+
+        }
+    }
+
+}
+
+export { CardStorage } ;
\ No newline at end of file
